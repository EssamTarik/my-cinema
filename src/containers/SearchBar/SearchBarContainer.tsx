import React, { useCallback } from 'react';
import debounce from 'lodash.debounce';
import { SearchBar } from '../../components';
import { IProps } from './interfaces';

const SearchBarContainer = ({ searchMovies }: IProps) => {
  const handleChange = debounce(useCallback((query: string) => searchMovies(query, 1), [searchMovies]), 300);

  return <SearchBar onChange={handleChange} />
}

export default SearchBarContainer;