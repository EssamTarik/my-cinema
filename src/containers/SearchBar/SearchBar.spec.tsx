import React from 'react';
import SearchBarContainer from './SearchBarContainer';
import { render, cleanup } from '@testing-library/react';
jest.mock('../../components', () => ({ SearchBar: () => <div /> }))

describe('SearchBarContainer', () => {
  afterEach(cleanup);
  const mockSearchMovies = jest.fn()

  it('renders', () => {
    const { asFragment } = render(<SearchBarContainer searchMovies={mockSearchMovies} />);
    expect(asFragment()).toMatchSnapshot()
  });
})