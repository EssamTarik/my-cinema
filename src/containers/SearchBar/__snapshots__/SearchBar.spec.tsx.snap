// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBarContainer renders 1`] = `
<DocumentFragment>
  <div />
</DocumentFragment>
`;
